app/Mage.php

```diff
@@ -865,7 +865,14 @@ public static function log($message, $level = null, $file = '', $forceLog = fals
 
         static $loggers = array();
 
+        $maxLogLevel = (int) self::getStoreConfig('dev/log/max_level');
+
         $level  = is_null($level) ? Zend_Log::DEBUG : $level;
+
+        if ($level > $maxLogLevel) {
```

Comment:
we should still respect developerMode here as higher priority, means $isDeveloperMode === true => log everything